src/xenia/kernel/XLiveAPI.cpp

```diff
@@ -191,20 +194,19 @@ void XLiveAPI::Init() {
   upnp_handler = new UPnP();
   mac_address_ = new MacAddress(GetMACaddress());
 
-  if (cvars::offline_mode) {
```

Comment:
The changes to lines 194-228 slightly change the logic so that system-link will function regardless of offline_mode's value. I was not able to test UPNP with these changes so that's an unknown. (My router does not support UPNP)